apps/server/src/modules/task/controller/dto/task.response.ts

```diff
@@ -1,31 +1,36 @@
 import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
 import { PaginationResponse, DecodeHtmlEntities } from '@shared/controller';
+import { TaskStatusResponse } from './task-status.response';
 
 /**
  * DTO for returning a task document via api.
  */
 export class TaskResponse {
-	constructor({ name, id, createdAt, updatedAt, status }: TaskResponse) {
-		this.name = name;
+	constructor({ id, name, createdAt, updatedAt, status }: TaskResponse) {
 		this.id = id;
+		this.name = name;
+		this.courseName = '';
```

Comment:
why not make/jkeep this argument optional ?

Reply:
we are using the empty string task name in the courses filter for tasks that don't belong to a course. yes, maybe the optional response property would be cleaner in terms of the api when there is no course reference.
**but** we would then have to map the tasks we get from the api to our client type. currently (as in all other stores as well) there is no mapping. if the property would be optional the types would not match:
https://github.com/hpi-schul-cloud/nuxt-client/blob/cbdd1854808fb35df8b3a6ca605cab5c9de79462/src/store/tasks.ts#L57

Reply:
why not have the prop optional in the client too?